.github/workflows/ci.yml

```diff
@@ -11,15 +11,38 @@ jobs:
   ci:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - name: Setup Node.js
+      - name: "Checkout SDK"
+        uses: actions/checkout@v2
+      - name: "Prettier Check"
+        uses: creyD/prettier_action@v4.0
+        with:
+          dry: True
+          prettier_options: --config .prettierrc --check **/*.ts
+      - name: "Checkout Erdstall Core"
+        uses: actions/checkout@v2
+        with:
+          ssh-key: ${{ secrets.SSH_PRIV_KEY }}
+          repository: "perun-network/erdstall-ext"
+          path: "erdstall"
+      - name: "Set up Go 1.x"
+        uses: actions/setup-go@v2
+        with:
+          go-version: ^1.17
+      - name: "Build walkthrough operator and store in environment"
+        working-directory: "erdstall"
+        run: |
+          go build -o wo ./cmd/walkthrough/main.go
+          echo "ERDSTALL_EXE=$( realpath wo )" >> $GITHUB_ENV
```

Comment:
I think you should move the whole Go stuff just before running `Test:Integration`. If any of the next steps fail, the whole Go setup was a waste.

And maybe we should even put the End-2-End Test into a separate `job`?

Reply:
That is a very good idea!

Reply:
As to whether or not we want a separate `job` for the end to end test:
Would this not require us to use something like `actions/upload-artifact` to avoid repulling all dependencies and rebuilding the project again or is there an easier way to share the data between the jobs?

If it is only doable by the mentioned gh action script then I would argue it is not really worth it to split. We could also argue about the possibility that the end2end test should stand completely for itself, s.t. it can be triggered on its own?

Reply:
We discussed to keep it sequentially in a single job.